GitCommands/StringExtensions.cs

```diff
@@ -189,6 +190,20 @@ public static string Quote(this string? s, string q = "\"")
             return string.IsNullOrEmpty(s) ? s : s.Quote();
         }
 
+        /// <summary>
+        /// Quotes this string for use as a command line argument.
+        /// </summary>
+        [Pure]
+        public static string QuoteForCommandLine(this string? s)
+        {
+            if (s is null)
+            {
+                return "";
+            }
+
+            return "\"" + (EnvUtils.RunningOnWindows() ? s.Replace("\"", "\"\"") : s.Replace("\\", "\\\\").Replace("\"", "\\\"").Replace("\'", "\\\'")) + "\"";
```

Comment:
Please add tests for this. Some comments could also help in understanding the chain of replaces.

Reply:
How does one design platform-specific tests? Since the quoting pattern varies by platform.

Reply:
Why use for bool though?
```cs
 public static string QuoteForCommandLine(this string s, bool runningOnWindows = true)
```

Reply:
It allows the test to provide an override, so the test can do checks on both the Windows and Posix implementations. See the additions to `GitCommandsTests/StringExtensionTests` to see how it is used.

This is basically a DI-equivalent. If not specified, it simply auto-detects the platform. That's why the default has to be `null`.

An alternative would be to DI `EnvUtils` itself. But that seems like overkill for just one bool.